src/components/plots/ClassifierKneePlot.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Vega } from 'react-vega';
+
+import EmptyPlot from './helpers/EmptyPlot';
+import generateSpec from '../../utils/plotSpecs/generateClassifierKneePlot';
+
+const ClassifierKneePlot = (props) => {
+  const {
+    config, expConfig, plotData, actions,
+  } = props;
+
+  const [plotSpec, setPlotSpec] = useState(config);
+
+  useEffect(() => {
+    if (config && plotData) {
```

Comment:
I think you can assume the config and plotData are defined no? I'm looking at `Classifier.jsx` line 108.

Reply:
We are actually doing this in other plots too (like `DoubletScoreHistogram.jsx`) although I agree that I would like remove these checks.

My guess is we have them because of the `MiniPlot` components which don't check that plotData is defined, I think it would be nice to also check in `MiniPlot` so that we can go ahead and make the plot component simpler as you suggested.

Reply:
A good separation of concerns is to make sure all the data we need is loaded in the index files and then pass them to the plot components. Less bugs and undefined variables to worry about.

Reply:
Sure, but the Miniplot component has the selectors in it, so that's when it can check on them.